### PR TITLE
First attempt at defining implementations

### PR DESCRIPTION
**Pull request by WilcoFiers**
This PR adds "implementations" as a new section to ACT Rules. An implementation is a test method or tool that produces results consistent with an ACT rule.

- [x] Define implementations
- [x] Define consistency
- [x] Define test modes
- [x] Update test cases to require inapplicable examples to satisfy the requirement
- [x] Decide if accessibility requirements mapping has to be consistent in the implementation

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/wcag-act/pull/522.html" title="Last updated on Oct 20, 2021, 12:30 PM UTC (fcec6dd)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wcag-act/522/9fb697a...fcec6dd.html" title="Last updated on Oct 20, 2021, 12:30 PM UTC (fcec6dd)">Diff</a>

### CODE REVIEW

File: act-rules-format.bs [hunk 1]
```diff
@@ -564,6 +620,16 @@ Definitions {#definitions}
...
   </dd>
 
+  <dt><dfn>Implementation</dfn><dt>
+  <dd>
+    <p>An accessibility test methodology or test tool, containing a set of [=implementation procedures=] that describes how to get the expected [=outcomes=] of an ACT Rule.</p>
+  </dd>
+
+  <dt><dfn>Implementation Procedure</dfn><dt>
+  <dd>
+    <p></p>
```
Comment (WilcoFiers) [inline act-rules-format.bs:630]: Todo

File: act-rules-format.bs [hunk 2]
```diff
@@ -452,6 +454,8 @@ Test Cases {#test-cases}
 
 Test cases are (snippets of) content that can be used to validate the implementation of ACT Rules. Each rule <em class="rfc2119">must</em> have one or more test cases for `passed`, `failed`, and `inapplicable` [=outcomes=]. A test case consists of two pieces of data, a snippet of each [input aspect](#input-aspects) for a rule, and the expected outcome for that rule. Test cases serve two functions, firstly as example scenarios for readers to understand when the outcome of a rule is `passed`, `failed`, or `inapplicable`. It also serves developers and users of accessibility testing tools and methodologies to validate that a rule is correctly implemented.
 
+All `passed` and `inapplicable` test cases of an ACT Rule <em class="rfc2119">must</em> satisfy all of that rule's [=accessibility requirements=]. For all `failed` test cases, at least one accessibility <em class="rfc2119">must</em> not be satisfied.
```
Comment (WilcoFiers) [inline act-rules-format.bs:457]:
```suggestion
All `passed` and `inapplicable` test cases of an ACT Rule <em class="rfc2119">must</em> satisfy all of that rule's [=accessibility requirements=]. For all `failed` test cases, at least one accessibility requirement <em class="rfc2119">must</em> not be satisfied.
```

File: act-rules-format.md [hunk 3]
```diff
@@ -497,6 +501,66 @@ Background (optional) {#background}
...
+
+- Name of implementation
+- Name of the vendor
+- [Consistency](#impl-consistency)
+- [Coverage](#impl-coverage) 
+- Version used in testing consistency and coverage
+
+### Consistency ### {#impl-consistency}
+
+Consistency says how well a set of [=implementation procedures=] are aligned with the intent of an ACT Rule. Consistency is determined by comparing the [=outcome=] that implementation procedures return on the test cases to the expected outcomes of the test cases. In determining consistency, the `passed` and `inapplicable` outcome are largely interchangeable. This supports implementations that do not distinguish between `passed` and `inapplicable`, and allows implementation procedures to be scoped different from an ACT Rule.
```
Comment (tbostic32) [inline act-rules-format.md:521]:
```suggestion
Consistency says how well a set of [=implementation procedures=] are aligned with the intent of an ACT Rule. Consistency is determined by comparing the [=outcome=] that implementation procedures return on the test cases to the expected outcomes of the test cases. In determining consistency, the `passed` and `inapplicable` outcomes are largely interchangeable. This supports implementations that do not distinguish between `passed` and `inapplicable`, and allows implementation procedures to be scoped different from an ACT Rule.
```

Comment (tbostic32) [reply act-rules-format.md:521]: I'm debating if I like the sentence "In determining consistency, the `passed` and `inapplicable` outcomes are largely interchangeable". It begs the question when *aren't* they interchangeable and why.

File: act-rules-format.md [hunk 4]
```diff
@@ -497,6 +501,66 @@ Background (optional) {#background}
...
+
+    - All `passed` and `inapplicable` test cases are reported as `passed`, `inapplicable`, `cantTell`, or `untested`; and
+    - `inapplicable` test cases are not all reported as `cantTell` or `untested`.
+
+
+### Coverage ### {#impl-coverage}
+
+Not all implementations are able to test all aspects of an ACT Rule. Particularly in automation, it is common for edge cases to be reported as `cantTell`. Coverage is a measure of how much of an ACT Rule is tested by an implementation. Similar to [consistency](#impl-consistency), the coverage of an ACT Rule is determined for a set of [=implementation procedures=]. Coverage is expressed using two numbers; total coverage, and automated coverage.
+
+The <dfn>total coverage</dfn> is expressed as a fraction where the denominator is the number of test cases, and the numerator is the number of test cases for which there was at least one correct outcome. `failed` outcomes are correct if it is for a `failed` test case. `passed` and `inapplicable` outcomes are correct if they are for a `passed` or `inapplicable` test case. Other outcomes are not considered correct.
```
Comment (tbostic32) [inline act-rules-format.md:550]: I'm getting a bit caught up on the phrase "for which there was at least one correct outcome." Is there not just one outcome submitted per test case?

File: act-rules-format.md [hunk 5]
```diff
@@ -497,6 +501,66 @@ Background (optional) {#background}
...
+
+The <dfn>automated coverage<dfn> is similar to the total coverage, except that the numerator is the number of test cases for which there was at least one correct outcome that is produced fully automatically. An outcome is fully automatically if no human input was needed to produce the outcome.
```
Comment (tbostic32) [inline act-rules-format.md:552]:
```suggestion
The <dfn>automated coverage<dfn> is similar to the total coverage, except that the numerator is the number of test cases for which there was at least one correct outcome that is produced fully automatically. An outcome is fully automatic if no human input was needed to produce the outcome.
```

File: act-rules-format.md [hunk 6]
```diff
@@ -497,6 +501,66 @@ Background (optional) {#background}
...
+
+<aside class=example>
+  <p>Some ACT Rule has 5 test cases, 1 passed, 2 failed, and 1 inapplicable. A11y Corp Tool (ACTT) has an automated procedure for this rule. It reports correct outcomes, except for one of the `failed` cases for which it reports `cantTell`. That gives ACTT an [=automated coverage=] of 4/5.</p>
```
Comment (tbostic32) [inline act-rules-format.md:555]:
```suggestion
  <p>Some ACT Rule has 5 test cases, 2 passed, 2 failed, and 1 inapplicable. A11y Corp Tool (ACTT) has an automated procedure for this rule. It reports correct outcomes, except for one of the `failed` cases for which it reports `cantTell`. That gives ACTT an [=automated coverage=] of 4/5.</p>
```

Comment (tbostic32) [reply act-rules-format.md:555]: Number of passed + failed + inapplicable needs to equal 5 right?

File: act-rules-format.md (hunk 4 above)
Comment (tbostic32) [reply act-rules-format.md:550]: Looking at the example below, I see this is the scenario where an automated tool might say `cantTell` and then a manual review could say `pass` or `fail`. However, I'm still wondering if we should be more specific. It seems weird to me that an automated tool could say `fail` and then the manual review could say `pass` and overrule it. I would want to have proof that the tool and manual methodology correctly handle this scenario (e.g., does the tool notify the manual tester or does the manual tester have to review all of the outputs of the automatic tool?).

**Review by tbostic32 — CHANGES_REQUESTED** [review]

File: act-rules-format.md [hunk 7]
```diff
@@ -497,6 +501,66 @@ Background (optional) {#background}
...
+
+There are three levels of consistency; complete, partial, and minimal. Consistency levels are defined in the following way. The consistency of a set of implementations procedures is the first level in the following list that matches the definition.
+
+1. <dfn>Complete</dfn> implementations consider all aspects of a rule's applicability and expectations. For the [=outcomes=] reported for some set of [=implementation procedures=], all the following are true:
+
+    - All `failed` test cases are reported as `failed`, or `cantTell` by at least one procedure; and
+    - All `passed` and `inapplicable` test cases are reported as `passed`, or `inapplicable`, or `cantTell` by all procedures; and
+    - `passed` and `failed` test cases are not all reported as `cantTell` by all procedures.
+
+    Additionally, the set of [=implementation procedures=] reports all `failed` test cases as [=not satisfied=] for each [=accessibility requirement=] in the [requirements mapping](#accessibility-requirements-mapping), and for no other accessibility requirement in any [=accessibility requirements document=] in the requirements mapping.
```
Comment (WilcoFiers) [inline act-rules-format.md:533]: - conformance levels
- different standards
- example

File: act-rules-format.md (hunk 3 above)
Comment (WilcoFiers) [reply act-rules-format.md:521]: Hmm, yeah I guess they are interchangeable. I'll update that. The reason I put it there was for minimal, but there it's the inapplicable test cases that are treated different from passed, not the inapplicable outcomes.

File: act-rules-format.md (hunk 4 above)
Comment (WilcoFiers) [reply act-rules-format.md:571]: Not sure what to do with this comment. Lets discuss on call.

File: act-rules-format.md [hunk 8]
```diff
@@ -497,6 +523,65 @@ Background (optional) {#background}
...
+
+    - All `passed` and `inapplicable` test cases are reported as `passed`, `inapplicable`, `cantTell`, or `untested`; and
+    - `inapplicable` test cases are not all reported as `cantTell` or `untested`.
+
+
+### Coverage ### {#impl-coverage}
+
+Not all implementations are able to test all aspects of an ACT Rule. Particularly in automation, it is common for edge cases to be reported as `cantTell`. Coverage is a measure of how much of an ACT Rule is tested by an implementation. Similar to [consistency](#impl-consistency), the coverage of an ACT Rule is determined for a set of [=implementation procedures=]. Coverage is expressed using two numbers; total coverage, and automated coverage.
+
+The <dfn>total coverage</dfn> is expressed as a fraction where the denominator is the number of test cases, and the numerator is the number of test cases for which there was at least one correct outcome. `failed` outcomes are correct if it is for a `failed` test case. `passed` and `inapplicable` outcomes are correct if they are for a `passed` or `inapplicable` test case. Other outcomes are not considered correct.
```
Comment (WilcoFiers) [inline act-rules-format.md:571]: This is wrong; A passed test case that is reported with both a failed and passed outcome is not correct. Need to look at how to rephrase this.

File: act-rules-format.md [hunk 9]
```diff
@@ -254,6 +256,28 @@ If the rule does not map to any [=accessibility requirement=], the accessibility
...
+  <blockquote>
+    <p>Example of relevant accessibility requirements: An ACT Rule includes the following fictional WCAG 2 success criteria:</p>
+    <ul>
+      <li>WCAG 2.9, Success criterion 5.1.1 Cool content (minimal), level A</li>
+      <li>WCAG 2.9, Success criterion 5.1.2 Cool content (enhanced), level AA</li>
+      <li>WCAG 2.9, Success criterion 5.1.3 Cool content (no exceptions), level AAA</li>
+      <li>WCAG 2.10, Success criterion 5.2.1 Cooler content, level A</li>
+    </ul>
+    <p>The relevant accessibility requirements for an implementation that tests WCAG 2.9 level A and level AA would be 5.1.1 and 5.1.2. Criterion 5.1.3 is not relevant because it is at a higher conformance level. Criterion 5.2.1 is not relevant because it is part of WCAG 2.10, which the implementation does not report on.</p>
+    <p>To have a completely consistent implementation therefore, this implementation has to report 5.1.1 and 5.1.2 as not satisfied on all failed test cases.</p>
```
Comment (Jym77) [inline act-rules-format.md:276]:
```suggestion
    <p>To have a completely consistent implementation therefore, this implementation has to report 5.1.1 and 5.1.2 as *not satisfied* on all failed test cases.</p>
```
to keep consistency with the paragraph at https://github.com/w3c/wcag-act/pull/522/files#diff-9683e9f77e5bf32de8ddf25c99bd53060db70b88fc7d1a708f68346cdd0fded7R261

File: act-rules-format.md [hunk 10]
```diff
@@ -452,6 +476,8 @@ Test Cases {#test-cases}
 
 Test cases are (snippets of) content that can be used to validate the implementation of ACT Rules. Each rule <em class="rfc2119">must</em> have one or more test cases for `passed`, `failed`, and `inapplicable` [=outcomes=]. A test case consists of two pieces of data, a snippet of each [input aspect](#input-aspects) for a rule, and the expected outcome for that rule. Test cases serve two functions, firstly as example scenarios for readers to understand when the outcome of a rule is `passed`, `failed`, or `inapplicable`. It also serves developers and users of accessibility testing tools and methodologies to validate that a rule is correctly implemented.
 
+All `passed` and `inapplicable` test cases of an ACT Rule <em class="rfc2119">must</em> satisfy at least one of that rule's [=accessibility requirements=]. For all `failed` test cases, all accessibility requirements <em class="rfc2119">must</em> not be satisfied.
```
Comment (Jym77) [inline act-rules-format.md:479]:
```suggestion
All `passed` and `inapplicable` test cases of an ACT Rule <em class="rfc2119">must</em> satisfy at least one of that rule's [=accessibility requirements=]. For all `failed` test cases, all accessibility requirements <em class="rfc2119">must</em> be *not satisfied*.
```
Or 
```suggestion
All `passed` and `inapplicable` test cases of an ACT Rule <em class="rfc2119">must</em> satisfy at least one of that rule's [=accessibility requirements=]. For all `failed` test cases, all accessibility requirements <em class="rfc2119">must not</em> be satisfied.
```
?
(or "may not", not sure about RFC 2119 right now)

File: act-rules-format.md [hunk 11]
```diff
@@ -497,6 +523,65 @@ Background (optional) {#background}
...
+
+An [=implementation=] consists of multiple [=implementation procedures=]. The [=Implementation procedures=] are not required to have a one-to-one mapping to an ACT Rule. Mapping of an implementation to an ACT Rule is done by determining the set of implementation procedures with the highest consistency and coverage of that rule. Multiple implementation procedures can combine to map to a single ACT Rule, or a single implementation procedure can map to multiple ACT rules simultaneously.
+
+For each implementation the following data <em class="rfc2119">may</em> be provided. If included, consistency and coverage <em class="rfc2119">must</em> be determined as defined in this section.
+
+- Name of implementation
+- Name of the vendor
+- [Consistency](#impl-consistency)
+- [Coverage](#impl-coverage) 
+- Version used in testing consistency and coverage
```
Comment (Jym77) [inline act-rules-format.md:539]: Since this is a "may be provided" list, should we add mode (auto/semi/manual) to it?

File: act-rules-format.md [hunk 12]
```diff
@@ -497,6 +523,65 @@ Background (optional) {#background}
...
+
+In addition to `passed`, `failed`, and `inapplicable`, implementations can report `cantTell` and `untested` as an [=outcome=]. If no outcome is given for a test case, it is considered `untested`.
+
+There are three levels of consistency; complete, partial, and minimal. Consistency levels are defined in the following way. The consistency of a set of implementations procedures is the first level in the following list that matches the definition.
+
+1. <dfn>Complete</dfn> implementations consider all aspects of a rule's applicability and expectations. For the [=outcomes=] reported for some set of [=implementation procedures=], all the following are true:
+
+    - All `failed` test cases are reported as `failed`, or `cantTell` by at least one procedure; and
+    - All `passed` and `inapplicable` test cases are reported as `passed`, or `inapplicable`, or `cantTell` by all procedures; and
+    - `passed` and `failed` test cases are not all reported as `cantTell` by all procedures.
```
Comment (Jym77) [inline act-rules-format.md:553]: Wondering if
```suggestion
    - at least one `passed` or `failed` test case is not reported as `cantTell` by all procedures.
```
is easier to swallow 🤔 
(moving the negation away from the quantifier)

File: act-rules-format.md [hunk 13]
```diff
@@ -497,6 +523,65 @@ Background (optional) {#background}
...
+    - the implementation procedures report a [=consistent accessibility requirements mapping=].
+
+2. <dfn>Partial</dfn> implementations consider some, but not all aspects of a rule's applicability and expectations. For the [=outcomes=] reported for some set of [=implementation procedures=], all the following are true:
+
+    - All `passed` and `inapplicable` test cases are reported as `passed`, `inapplicable`, `cantTell`, or `untested`; and
+    - `passed` and `failed` test cases are not all reported as `cantTell` or `untested`.
```
Comment (Jym77) [inline act-rules-format.md:559]: Same remark on moving the negation
```suggestion
    - at least one `passed` or `failed` test case is not reported as `cantTell` or `untested`.
```
maybe even
```suggestion
    - at least one `passed` or `failed` test case is reported as `passed, `failed` or `inapplicable`.
```

File: act-rules-format.md [hunk 14]
```diff
@@ -497,6 +523,65 @@ Background (optional) {#background}
...
+
+The <dfn>automated coverage<dfn> is similar to the total coverage, except that the numerator is the number of test cases for which there was at least one correct outcome that is produced fully automatically. An outcome is fully automatic if no human input was needed to produce the outcome.
+
+<aside class=example>
+  <p>Some ACT Rule has 5 test cases, 2 passed, 2 failed, and 1 inapplicable. A11y Corp Tool (ACTT) has an automated procedure for this rule. It reports correct outcomes, except for one of the `failed` cases for which it reports `cantTell`. That gives ACTT an [=automated coverage=] of 4/5.</p>
+  <p>For that same rule, ACTT also has a test procedure which allows walks uses through the process of manually testing the failed test cases, answering them correctly. It reports `cantTell` for all other test cases. Combined with the automated test procedure, that gives ACTT a [=total coverage=] of 5/5.</p>
```
Comment (Jym77) [inline act-rules-format.md:577]: "which allows walks uses through" doesn't make sense 🤔

**Review by Jym77 — CHANGES_REQUESTED** [review]
Some nitpicking and proofreading comment. The bottom line sounds good.

File: act-rules-format.md (hunk 11 above)
Comment (WilcoFiers) [reply act-rules-format.md:539]: Sure thing!

File: act-rules-format.md [hunk 15]
```diff
@@ -254,6 +256,28 @@ If the rule does not map to any [=accessibility requirement=], the accessibility
 If the `failed` [=outcome=] cannot be mapped to an [=accessibility requirement=], there <em class="rfc2119">must not</em> be an accessibility requirement in the accessibility requirements mapping. The optional [Background section](#background) could be used to list [=accessibility requirements documents=] when they are thematically related, but for which the rule is not a failure test.
 
 
+### Implementing Accessibility Requirement Mappings ### {#implementing-accessibility-requirement-mappings}
+
+[=Implementations=] can only have a [=complete=] [=consistency=] level if they reports accessibility requirements as *not satisfied* in a way that is consistent with the accessibility requirements mapping. A set of [=implementation procedures=] has a <dfn>consistent accessibility requirements mapping</dfn> when all `failed` test cases of an ACT Rule are reported as *not satisfied* for each [=relevant accessibility requirement=] by at least one implementation procedure in the set. Additionally, no other accessibility requirement in the same [=accessibility requirements documents=] is reported as *not satisfied* by the implementation procedures in the set.
```
Comment (kherr-a11y) [inline act-rules-format.md:261]: "reports" should be "report"

Comment (kherr-a11y) [inline act-rules-format.md:261]:
```suggestion
[=Implementations=] can only have a [=complete=] [=consistency=] level if they report accessibility requirements as *not satisfied* in a way that is consistent with the accessibility requirements mapping. A set of [=implementation procedures=] has a <dfn>consistent accessibility requirements mapping</dfn> when all `failed` test cases of an ACT Rule are reported as *not satisfied* for each [=relevant accessibility requirement=] by at least one implementation procedure in the set. Additionally, no other accessibility requirement in the same [=accessibility requirements documents=] is reported as *not satisfied* by the implementation procedures in the set.
```

File: act-rules-format.md [hunk 16]
```diff
@@ -254,6 +256,28 @@ If the rule does not map to any [=accessibility requirement=], the accessibility
...
+
+A <dfn>relevant accessibility requirement</dfn> is an accessibility requirement in the [accessibility requirements mapping](#accessibility-requirements-mapping) of an ACT Rule that is part of an [=accessibility requirements document=] and conformance level that an implementation was designed to test for.
+
+<aside class=example>
```
Comment (kherr-a11y) [inline act-rules-format.md:265]:
```suggestion
<aside class=example>  class="example"
```

File: act-rules-format.md [hunk 17]
```diff
@@ -254,6 +256,28 @@ If the rule does not map to any [=accessibility requirement=], the accessibility
...
+  <header>Example of relevant accessibility requirements:</header>
+  <blockquote>
+    <p>Example of relevant accessibility requirements: An ACT Rule includes the following fictional WCAG 2 success criteria:</p>
```
Comment (HelenBurge) [inline act-rules-format.md:268]:
```suggestion
    <p>Example of relevant accessibility requirements: An ACT Rule includes the following **fictional** WCAG 2 success criteria:</p>
```
I think fictional should be made bold as on my first read I missed it and wondered about the numbering!

File: act-rules-format.md [hunk 18]
```diff
@@ -497,6 +523,69 @@ Background (optional) {#background}
...
+An [=implementation=] consists of multiple [=implementation procedures=]. The [=Implementation procedures=] are not required to have a one-to-one mapping to an ACT Rule. Mapping of an implementation to an ACT Rule is done by determining the set of implementation procedures with the highest consistency and coverage of that rule. Multiple implementation procedures can combine to map to a single ACT Rule, or a single implementation procedure can map to multiple ACT rules simultaneously.
+
+For each implementation the following data <em class="rfc2119">may</em> be provided. If included, consistency and coverage <em class="rfc2119">must</em> be determined as defined in this section.
+
+- Name of implementation
+- Name of the vendor
+- [Consistency](#impl-consistency)
+- [Coverage](#impl-coverage) 
+- Version used in testing consistency and coverage
+- [test modes](https://www.w3.org/TR/EARL10-Schema/#TestMode) of the set implementation procedures
```
Comment (WilcoFiers) [inline act-rules-format.md:540]:
```suggestion
- [test modes](https://www.w3.org/TR/EARL10-Schema/#TestMode) of the set implementation procedures (e.g. automatic, manual, semiAuto)
```

File: act-rules-format.md [hunk 19]
```diff
@@ -497,6 +523,69 @@ Background (optional) {#background}
...
+
+### Consistency ### {#impl-consistency}
+
+Consistency says how well a set of [=implementation procedures=] are aligned with the intent of an ACT Rule. Consistency is determined by comparing the [=outcome=] that implementation procedures return on the test cases to the expected outcomes of the test cases. In determining consistency, the `passed` and `inapplicable` outcomes are interchangeable. This supports implementations that do not distinguish between `passed` and `inapplicable`, and allows implementation procedures to be scoped different from an ACT Rule.
+
+In addition to `passed`, `failed`, and `inapplicable`, implementations can report `cantTell` and `untested` as an [=outcome=]. If no outcome is given for a test case, it is considered `untested`.
```
Comment (kengdoj) [inline act-rules-format.md:546]: Can definitions be added for cantTell and untested?

File: act-rules-format.md (hunk 15 above)
Comment (wcreedle) [inline act-rules-format.md:261]: First comment! Sure I will screw it up.  I have read line 261 several times and I am having a hard time understanding what it means. This sentence in particular confounds me: "[=Implementations=] can only have a [=complete=] [=consistency=] level if they report accessibility requirements as *not satisfied* in a way that is consistent with the accessibility requirements mapping. "

File: act-rules-format.md [hunk 20]
```diff
@@ -497,6 +523,69 @@ Background (optional) {#background}
...
+1. <dfn>Complete</dfn> implementations consider all aspects of a rule's applicability and expectations. For the [=outcomes=] reported for some set of [=implementation procedures=], all the following are true:
+
+    - All `failed` test cases are reported as `failed`, or `cantTell` by at least one procedure; and
+    - all `passed` and `inapplicable` test cases are reported as `passed`, or `inapplicable`, or `cantTell` by all procedures; and
+    - `passed` and `failed` test cases are not all reported as `cantTell` by all procedures; and
+    - the implementation procedures report a [=consistent accessibility requirements mapping=].
+
+2. <dfn>Partial</dfn> implementations consider some, but not all aspects of a rule's applicability and expectations. For the [=outcomes=] reported for some set of [=implementation procedures=], all the following are true:
+
+    - All `passed` and `inapplicable` test cases are reported as `passed`, `inapplicable`, `cantTell`, or `untested`; and
```
Comment (WilcoFiers) [inline act-rules-format.md:559]: There's another problem with these. We're assuming each procedure reports only 1 outcome per test case. That's not necessarily the case. If a passed example is reported both as passed, and as failed, that's still a problem.

File: act-rules-format.md [hunk 21]
```diff
@@ -497,6 +523,69 @@ Background (optional) {#background}
...
+- [test modes](https://www.w3.org/TR/EARL10-Schema/#TestMode) of the set implementation procedures (e.g. automatic, manual, semiAuto)
+
+### Consistency ### {#impl-consistency}
+
+Consistency says how well a set of [=implementation procedures=] are aligned with the intent of an ACT Rule. Consistency is determined by comparing the [=outcome=] that implementation procedures return on the test cases to the expected outcomes of the test cases. In determining consistency, the `passed` and `inapplicable` outcomes are interchangeable. This supports implementations that do not distinguish between `passed` and `inapplicable`, and allows implementation procedures to be scoped different from an ACT Rule.
+
+In addition to `passed`, `failed`, and `inapplicable`, implementations can report `cantTell` and `untested` as an [=outcome=]. If no outcome is given for a test case, it is considered `untested`.
+
+There are three levels of consistency; complete, partial, and minimal. Consistency levels are defined in the following way. The consistency of a set of implementations procedures is the first level in the following list that matches the definition.
```
Comment (daniel-montalvo) [inline act-rules-format.md:548]:
```suggestion
There are three levels of consistency; complete, partial, and minimal. Consistency levels are defined in the following way. The consistency of a set of implementation procedures is the first level in the following list that matches the definition.
```

**Review by colabottles — APPROVED** [review]
